akka-javasdk/src/main/java/akka/javasdk/agent/RemoteMcpTools.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.javasdk.agent;
+
+import akka.annotation.DoNotInherit;
+import akka.http.javadsl.model.HttpHeader;
+import akka.javasdk.impl.agent.RemoteMcpToolsImpl;
+
+import java.util.function.Predicate;
+
+/**
+ * Access to tools from one remote MCP server.
+ *
+ * <p>Not for user extension, create instances using {@link #fromServer(String)}.
+ */
+@DoNotInherit
+public interface RemoteMcpTools {
+
+  /**
+   * @param serverUri A URI to the remote MCP HTTP server, for example "https://example.com/sse" or
+   *     "https://example.com/mcp
+   */
+  static RemoteMcpTools fromServer(String serverUri) {
+    return new RemoteMcpToolsImpl(serverUri);
+  }
+
+  /**
+   * Define a filter to select what discovered tool names are passed on to the chat model. Names
+   * that are filtered will not be described to the model and will not allow calls.
+   */
+  RemoteMcpTools allowToolNames(Predicate<String> toolNameFilter);
+
+  /**
+   * Specify an interceptor that has the capability to allow or deny calls (by throwing an
+   * exception) and also to modify and filter input to the MCP server tool.
+   */
+  RemoteMcpTools withToolInterceptor(ToolInterceptor interceptor);
```

Comment:
An alternative could be to have separate `withToolRequestInterceptor` and `withToolResponseInterceptor`, which would allow for lambdas. I don't know which is better.